test/install.c

```diff
@@ -1130,6 +1151,47 @@ static void install_test_bundle_hook_post_install(InstallFixture *fixture,
 	g_free(bundlepath);
 }
 
+/* Test with already mounted slot */
+static void install_test_already_mounted(InstallFixture *fixture,
+		gconstpointer user_data)
+{
+	gchar *bundlepath, *mountprefix, *hookfilepath;
+	RaucInstallArgs *args;
+	GError *ierror = NULL;
+	gboolean res;
+
+	/* needs to run as root */
+	if (!test_running_as_root())
+		return;
+
+	/* Set mount path to current temp dir */
+	mountprefix = g_build_filename(fixture->tmpdir, "mount", NULL);
+	g_assert_nonnull(mountprefix);
+	r_context_conf()->mountprefix = mountprefix;
+	r_context();
+
+	bundlepath = g_build_filename(fixture->tmpdir, "bundle.raucb", NULL);
+	g_assert_nonnull(bundlepath);
+
+	hookfilepath = g_build_filename(fixture->tmpdir, "bootloader", "hook-install-mounted", NULL);
+	g_assert_nonnull(hookfilepath);
+	g_assert_false(g_file_test(hookfilepath, G_FILE_TEST_EXISTS));
+
+	args = install_args_new();
+	args->name = g_strdup(bundlepath);
+	args->notify = install_notify;
+	args->cleanup = install_cleanup;
+	res = do_install_bundle(args, &ierror);
+	g_assert_no_error(ierror);
+	g_assert_true(res);
+
+	g_assert_true(g_file_test(hookfilepath, G_FILE_TEST_IS_REGULAR));
+
+	args->status_result = 0;
```

Comment:
Not really useful for anything but other tests also do this, thus maybe something to clean up separately.

Reply:
Yes, that should be a separate PR.